index.js

```diff
@@ -2,268 +2,299 @@
     stream = net.createConnection(opts.port, endpoint);
   }
 
   function onSecure() {
-    // let's just crash if we are not secure
     if (!stream.authorized) {
-      logDebug('Connection is not secure!');
-      throw new Error('secure connection not authorized');
+      // let's just crash if we are not secure
+      throw new Error('Secure connection is not authorized');
     }
+    LOGGER.info('Connection is secure');
   }
 
   return stream;
 }
 
 
 function start(opts) {
-  debugLogging = opts.debug;
-  const logsToken = opts.logstoken || opts.token;
-  const statsToken = opts.statstoken || opts.token;
-  const eventsToken = opts.eventstoken || opts.token;
   let out;
   let noRestart = () => void 0;
 
   const filter = through.obj(function (obj, enc, cb) {
-    logDebug(`Got an event with encoding "${enc}":`, obj);
+    LOGGER.debug(`Got an event with encoding "${enc}"`, obj);
 
+    LOGGER.debug('Enriching obj with --add')
     obj = addAll(opts.add, obj);
+    function addAll(proto, obj) {
+      LOGGER.debug('Checking proto', proto);
+      if (!proto) {
+        return;
+      }
+  
+      const newObj = {...obj};
+  
+      for (const key in proto) {
+        if (proto.hasOwnProperty(key)) {
+          LOGGER.debug(`Adding key "${key}" value "${proto[key]}"`);
+          newObj[key] = proto[key];
+        }
+      }
+  
+      return newObj;
+    }
+
+    LOGGER.debug('Getting correct token for obj...')
     const token = (() => {
-      const {
-        line,
-        type,
-        stats
-      } = obj;
-
-      if (line) {
-        logDebug('Using logs token:', logsToken);
-        return logsToken;
-      } else if (type) {
-        logDebug('Using events token:', eventsToken);
-        return eventsToken;
-      } else if (stats) {
-        logDebug('Using stats token:', statsToken);
-        return statsToken;
+      if (obj.line) {
+        LOGGER.debug('Using logs token:', opts.logstoken);
+        return opts.logstoken;
+      } else if (obj.type) {
+        LOGGER.debug('Using events token:', opts.eventstoken);
+        return opts.eventstoken;
+      } else if (obj.stats) {
+        LOGGER.debug('Using stats token:', opts.statstoken);
+        return opts.statstoken;
+      } else {
+        LOGGER.debug('Unable to figure out correct token to use, skipping log', obj);
       }
     })();
 
     if (token) {
-      logDebug('Prepending log token:', token);
+      LOGGER.debug('Stringifying object and prepending log token:', token);
 
       this.push(token);
       this.push(' ');
       this.push(JSON.stringify(obj));
       this.push('\n');
     }
 
-    logDebug('Finished processing event:', obj);
+    LOGGER.debug('Finished processing event, calling callback');
     cb();
   });
 
+  LOGGER.debug('Getting all containers events...')
   const events = allContainers(opts);
   let streamsOpened = 0;
   opts.events = events;
 
+  LOGGER.debug('Creating log hose');
```

Comment:
except you're actually not doing that. the line below is a function definition, we're not calling it.